qt/widgets/common/src/FunctionModel.cpp

```diff
@@ -487,6 +489,37 @@ void FunctionModel::updateGlobals() {
   }
 }
 
+void FunctionModel::setResolutionFromWorkspace(IFunction_sptr fun) {
+  auto n = fun->getNumberDomains();
+  if (n > 1) {
+    for (size_t index = 0; index < n; index++) {
+      setResolutionFromWorkspace(fun->getFunction(index));
+    }
+  } else {
+    if (fun->hasAttribute("f0.Workspace")) {
+      std::string wsName = fun->getAttribute("f0.Workspace").asString();
+      if (AnalysisDataService::Instance().doesExist(wsName)) {
+        const auto ws =
```

Comment:
I'm finding this a little difficult to parse due to the heavy nesting. Could we maybe extract out another method
```cpp
setResolutionFromWorkspace(const MatrixWorkspace & workspace)
```

that could be called as

```cpp
setResolutionFromWorkspace(AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(wsName));
```
?